extended/src/main/java/io/kubernetes/client/extended/event/legacy/ObjectReferenceResolvingEventRecorder.java

```diff
@@ -0,0 +1,125 @@
+package io.kubernetes.client.extended.event.legacy;
+
+import com.google.common.base.Strings;
+import io.kubernetes.client.extended.event.EventType;
+import io.kubernetes.client.openapi.models.*;
+import io.kubernetes.client.util.ObjectAccessor;
+import io.kubernetes.client.util.TypeAccessor;
+import io.kubernetes.client.util.exception.ObjectMetaReflectException;
+import io.kubernetes.client.util.exception.TypeMetaReflectException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ObjectReferenceResolvingEventRecorder implements EventRecorder {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(ObjectReferenceResolvingEventRecorder.class);
+
+  public ObjectReferenceResolvingEventRecorder(BlockingQueue<V1Event> queue) {
+    this.pendingEventQueue = queue;
+  }
+
+  private BlockingQueue<V1Event> pendingEventQueue;
+
+  @Override
+  public void event(Object object, EventType t, String reason, String format, String... args) {
+    event(object, new HashMap<>(), t, reason, format, args);
+  }
+
+  @Override
+  public void event(
+      Object object,
+      Map<String, String> attachedAnnotation,
+      EventType t,
+      String reason,
+      String format,
+      String... args) {
+    generateEvent(object, attachedAnnotation, t, reason, String.format(format, args));
+  }
+
+  @Override
+  public void event(
+      V1ObjectReference ref, EventType t, String reason, String format, String... args) {
+    event((Object) ref, t, reason, format, args);
+  }
+
+  @Override
+  public void event(
+      V1ObjectReference ref,
+      Map<String, String> attachedAnnotation,
+      EventType t,
+      String reason,
+      String format,
+      String... args) {
+    event((Object) ref, attachedAnnotation, t, reason, format, args);
+  }
+
+  private void generateEvent(
+      Object eventRefObject,
+      Map<String, String> annotations,
+      EventType t,
+      String reason,
+      String message) {
+    // get reference
```

Comment:
delete this? It doesn't seem to help much in understanding the code.

Reply:
deleted extra comments